langroid/agent/special/neo4j/neo4j_chat_agent.py

```diff
@@ -398,7 +400,8 @@ def graph_schema_tool(
 
     def _init_tools_sys_message(self) -> None:
         """Initialize message tools used for chatting."""
-        self.tried_schema = False
+        if not self.tried_schema:
```

Comment:
superfluous

Reply:
I had a case where DB exists, so `tried_schema` will be true after calling `graph_schema_tool` from `_initialize_db`. But when `_init_tools_sys_message` is called, which is immediately after `_init_tools_sys_message`, it sets `tried_schema` to false at [this line](https://github.com/langroid/langroid/blob/main/langroid/agent/special/neo4j/neo4j_chat_agent.py#L401)

Reply:
but the code is superfluous - if it's false, you're setting it to false, which is superfluous